dp/knapsack/Solution.java

```diff
@@ -37,32 +32,53 @@ public int maxProfit(int[] weights, int[] prices, int capacity) {
 class Memoization implements Solution {
 
     private int[][] profitBySelectAndCapacity;
-    private boolean selected[];
 
-    private int maxP(int[] weights, int[] prices, int capacity, int cumulativeProfit, int i) {
```

Comment:
It turns out that passing return value as a parameter could be not only much slower but also error-prone (error for this case).